http3/roundtrip.go

```diff
@@ -25,6 +26,11 @@ type roundTripCloser interface {
 	io.Closer
 }
 
+type roundTripCloserWithCount struct {
+	roundTripCloser
+	useCount atomic.Int64
```

Comment:
Both `getClient` and `CloseIdleConnections` run with the `Roundtripper.mutex` locked. Would it be sufficient to make this an `int`?

Reply:
https://github.com/quic-go/quic-go/blob/b6c36841a5e5063864b628dd2beab780a8d043f2/http3/roundtrip.go#L152
We reduce the count here, out of the lock. Unless we add a lock here, or we should use atomic.

Reply:
Makes sense, thank you!